cabal-install/Distribution/Solver/Types/ComponentDeps.hs

```diff
@@ -119,6 +121,17 @@ insert comp a = ComponentDeps . Map.alter aux comp . unComponentDeps
     aux Nothing   = Just a
     aux (Just a') = Just $ a `mappend` a'
 
+-- | Zip two 'ComponentDeps' together by 'Component', using 'mempty'
+-- as the neutral element when a 'Component' is present only in one.
+zip :: (Monoid a, Monoid b) => ComponentDeps a -> ComponentDeps b -> ComponentDeps (a, b)
+zip (ComponentDeps d1) (ComponentDeps d2) =
+    ComponentDeps $
+      Map.mergeWithKey
```

Comment:
unfortunately, `mergeWithKey` is available only since `containers-0.5` (and GHC 7.4 was still on containers-0.4)